src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java

```diff
@@ -74,148 +67,6 @@ void setUp() {
-            tcAttributes.put(RoleAttributeDefinition.JURISDICTION.value(), IA_JURISDICTION);
-            tcAttributes.put(RoleAttributeDefinition.CASE_TYPE.value(), IA_CASE_TYPE);
-            tcAttributes.put(RoleAttributeDefinition.CASE_ID.value(), caseId);
-            tcAttributes.put(LOCATION.name(), "1");
-        }
-
-        RoleAssignment roleAssignment = RoleAssignment
-            .builder()
-            .roleName("tribunal-caseworker")
-            .roleType(RoleType.ORGANISATION)
-            .classification(Classification.PUBLIC)
-            .beginTime(LocalDateTime.now().minusYears(1))
-            .endTime(LocalDateTime.now().plusYears(1))
-            .grantType(grantType)
-            .attributes(tcAttributes)
-            .build();
-        roleAssignments.add(roleAssignment);
-
-        AccessControlResponse assignerAccessControlResponse = new AccessControlResponse(null, roleAssignments);
-
-        List<PermissionTypes> assignerPermissionsRequired = new ArrayList<>();
-        assignerPermissionsRequired.add(PermissionTypes.MANAGE);
-
-        final Optional<TaskResource> task = cftQueryService.getTask(
-            taskId,
-            assignerAccessControlResponse.getRoleAssignments(),
-            assignerPermissionsRequired
-        );
-        Assertions.assertThat(task.isPresent()).isTrue();
-        Assertions.assertThat(task.get().getTaskId()).isEqualTo(taskId);
-        Assertions.assertThat(task.get().getCaseId()).isEqualTo(caseId);
-
-        List<RoleAssignment> assignerRoleAssignments = new ArrayList<>();
-        RoleAssignment assignerRoleAssignment = RoleAssignment
-            .builder()
-            .roleName("senior-tribunal-caseworker")
-            .roleType(RoleType.ORGANISATION)
-            .classification(Classification.PUBLIC)
-            .beginTime(LocalDateTime.now().minusYears(1))
-            .endTime(LocalDateTime.now().plusYears(1))
-            .grantType(grantType)
-            .attributes(tcAttributes)
-            .build();
-        assignerRoleAssignments.add(assignerRoleAssignment);
-
-        AccessControlResponse assigneeAccessControlResponse = new AccessControlResponse(
-            null,
-            assignerRoleAssignments
-        );
-        List<PermissionTypes> assigneePermissionsRequired = new ArrayList<>();
-        assigneePermissionsRequired.add(OWN);
-        assigneePermissionsRequired.add(EXECUTE);
-
-        final Optional<TaskResource> assignerTask = cftQueryService.getTask(
-            taskId,
-            assigneeAccessControlResponse.getRoleAssignments(),
-            assigneePermissionsRequired
-        );
-        Assertions.assertThat(assignerTask.isPresent()).isTrue();
-        Assertions.assertThat(assignerTask.get().getTaskId()).isEqualTo(taskId);
-        Assertions.assertThat(assignerTask.get().getCaseId()).isEqualTo(caseId);
-    }
-
-    @Test
-    void should_retrieve_a_task_to_assign_challenged() {
```

Comment:
Looks like we have removed this because it's IA centric. Does that mean we have the WA equivalent?

Reply:
wa/CftQueryServiceAssignTaskTest 
should_retrieve_a_task_when_grant_type_challenged